### PR TITLE
Add react 19 codemods and recommend `codemod` command

### PR DESCRIPTION
**Pull request by DmytroHryshyn**
- adds 7 react 19 codemods
- adds ability to run ts transfroms from `react-codemod` CLI
### Running with `react-codemod`
```
npx react-codemod remove-context-provider <path>
npx react-codemod replace-string-ref <path>
npx react-codemod remove-forward-ref <path>
npx react-codemod replace-use-form-state <path>
npx react-codemod use-context-hook <path>
npx react-codemod remove-memoization <path>
npx react-codemod replace-act-import <path>
```
### Running with `codemod`
```
npx codemod react/19/remove-context-provider --target <path>
npx codemod react/19/replace-string-ref --target <path>
npx codemod react/19/remove-forward-ref --target <path>
npx codemod react/19/replace-use-form-state --target <path>
npx codemod react/19/use-context-hook --target <path>
npx codemod react/19/remove-memoization --target <path>
npx codemod react/19/replace-act-import --target <path>
```

### CODE REVIEW

**Comment by alexbit-codemod** [comment]
This PR contributes 6 new codemods to `react-codemods` repository designed to streamline migration to React v19. In the root folder of v19 repo, there exists a README file that introduces Codemod CLI and how to run codemods both via Codemod CLI and react-codemod. in addition to that, each codemod has its own folder and for each codemod there exists a README that explains the transformation logic.

h/t @r4zendev for creating the React 19 codemods.

**Comment by alexbit-codemod** [comment]
@acdlite , @rickhanlonii who can review this? i would like us to merge this PR, invite a few early adopters before we market it more broadly. thank you.

**Comment by alexbit-codemod** [comment]
```
pnpm i -g codemod

codemod react/19/ref-to-arrow-function
codemod react/19/remove-forward-ref
codemod react/19/replace-use-form-state
codemod react/19/use-context-hook
codemod react/19/remove-memoization
codemod react/19/replace-act-import
```

**Comment by rickhanlonii** [comment]
@DmytroHryshyn can you allow edits to the PR?

**Comment by DmytroHryshyn** [comment]
> @DmytroHryshyn can you allow edits to the PR?

Sure, added you to the repo @rickhanlonii

**Comment by josephsavona** [comment]
We do not recommend removing useMemo and use callback while the compiler is still experimental. The compiler may skip components for various reasons (violations of React’s rules, as-yet-unsupported syntax, etc) and a blanket code of to remove memoization could cause problems. 

We strongly recommend waiting until the compiler is stable. We plan to provide a script that will only remove manual memoization where the compiler is running, skipping components/hooks where the compiler would bail out.

**Comment by alexbit-codemod** [comment]
noted. we will remove it soon. thanks for flagging.

**Review by rickhanlonii — APPROVED** [review]
Thanks!

File: README.md [hunk 1]
```diff
@@ -1,33 +1,93 @@
...
-   * use the `--dry` option for a dry-run and use `--print` to print the output for comparison
 
-This will start an interactive wizard, and then run the specified transform.
+We recommend using the [`codemod`](go.codemod.com/react-codemod) command for improved experience and support.
+
+`npx codemod <framework>/<version>/<transform> --target <path> [...options]`
+* `transform` - name of transform, see available transforms below.
+* `path` - directory to transform
+
+Check [codemod docs](https://go.codemod.com/cli-docs]) for the full list of available commands.
```
Comment (mohab-sameh) [inline README.md:15]: @rickhanlonii Typo: extra `]` in url.

File: README.md [hunk 2]
```diff
@@ -152,87 +201,15 @@ The wizard will ask for 1 option -
...
+## Support and Contributing
 
-The scripts in this repository are provided in the hope that they are useful,
-but they are not officially maintained, and we generally will not fix
-community-reported issues. They are a collection of scripts that were previously
-used internally within Facebook or were contributed by the community, and we
-rely on community contributions to fix any issues discovered or make any
-improvements. If you want to contribute, you're welcome to submit a pull
-request.
+The scripts in this repository are maintained by the React team in collaboration with the [codemods.com](https://codemods.com) team.
```
Comment (mohab-sameh) [inline README.md:209]: @rickhanlonii Typo: Extra 's'. should be codemod.com.

**Review by mohab-sameh — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (rickhanlonii) [reply README.md:15]:
```suggestion
Check [codemod docs](https://go.codemod.com/cli-docs) for the full list of available commands.
```

File: README.md [hunk 3]
```diff
@@ -1,33 +1,93 @@
...
+All codemods, whether you use the `codemod` CLI command or `react-codemod`, are free and open source, with the source code available in this repository.
 
 ### Usage
-`npx react-codemod <transform> <path> [...options]`
-   * `transform` - name of transform, see available transforms below.
-   * `path` - files or directory to transform
-   * use the `--dry` option for a dry-run and use `--print` to print the output for comparison
 
-This will start an interactive wizard, and then run the specified transform.
+We recommend using the [`codemod`](go.codemod.com/react-codemod) command for improved experience and support.
```
[the remark this answers is not in the document]
Comment (rickhanlonii) [reply README.md:9]: I don't have a preference, that was what the codemod team added in a previous commit on this PR

File: README.md (hunk 2 above)
Comment (rickhanlonii) [reply README.md:209]:
```suggestion
The scripts in this repository are maintained by the React team in collaboration with the [codemod.com](https://codemod.com) team.
```

**Review by mohab-sameh — APPROVED** [review]

**Comment by rickhanlonii** [comment]
Thanks!

**Comment by OliverJAsh** [comment]
> We do not recommend removing useMemo and use callback while the compiler is still experimental.

Is this still the case? From [the docs](https://react.dev/learn/react-compiler/introduction#what-should-i-do-about-usememo-usecallback-and-reactmemo):

> If you are using React Compiler, [useMemo](https://react.dev/reference/react/useMemo), [useCallback](https://react.dev/reference/react/useCallback), and [React.memo](https://react.dev/reference/react/memo) can be removed.

If the recommendation has changed, perhaps it's worth bringing back the codemod to remove memoization?

**Comment by alexbit-codemod** [comment]
`npx codemod@latest react/19/remove-memoization` ?

btw, this and other codemods here in the registry: https://app.codemod.com/registry/react/19/remove-memoization

**Pull request merged**